src/ContainerResolver.php

```diff
@@ -98,18 +102,31 @@ private function resolveMvcContainer(string $path): ContainerInterface
             Assert::isMap($appConfig);
         }
 
-        $servicesConfig = $appConfig['service_manager'] ?? [];
-        Assert::isMap($servicesConfig);
+        Assert::classExists(Application::class);
+
+        if ($appConfig['laminas-cli']['bootstrap_mvc_application'] ?? false) {
+            // initialize & bootstrap MVC Application
+            $mvcApplication = Application::init($appConfig);
```

Comment:
The current class is a _container resolver_ and not an _application initializer_. Maybe another [listener for the console application](https://github.com/laminas/laminas-cli/blob/97930db3fd2550da0374a3ccbdc7f9b9d6177319/src/ApplicationProvisioner.php#L39) would be a better place for this task. 🤔

Reply:
I'm not arguing here, but couldn't a similar argument currently be made regarding the whole _module loading_?

Reply:
Without loading the modules, you will not get the container configuration from the individual modules, which means that the entire container will not be resolved correctly.

Reply:
Besides my previous comment, which `ConsoleEvent`  could be used for initializing the `MvcApplication`?

Reply:
You can find the available events with description here: https://symfony.com/doc/current/components/console/events.html

Reply:
You're absolutely right. I did not do this to keep it BC.

- `loadModules()` is already it this place and is part of `::init()`, too
- moving the old code (the half App initialization) to a different place, changes the execution order and might cause a BC break
- separating the "old-way" from "the-new" way to two separate places would require checking the option in two places ... that's a dangerous thing to do, imo.  And changes execution order, too, in case you wan't to use the ::init().

This being said: In a release that is allowed to break BC I agree that the whole MVC Init should be outside of the Resolver. ( Including `loadModules()`, imo ). But with this there would be a lot more to do.  

For this quick fix, the best is to keep the footprint small and change as little as possible.

Reply:
And please note, I wrote "maybe". That means it is not the solution or even the correct solution. _That was just the first thought that came to my mind when I saw the names._ 😄

Reply:
Totally legit. Any thoughts are welcome. :-)  

That's what I can offer for now. I'd like to write a more clean solution but this will not happen within the next few days.  Maybe someone else can hop on earlier.